### DISCUSSION TITLE
Machine Learning Project

### DISCUSSION THREAD

**Discussion by laurenlarson** [williamgilpin/cphy#26] [projects]
This is a thread for Lauren, Bryce, Bryan, and Josey to work on our machine learning project.
@williamgilpin and @abao1999  We discussed after class some resources, and I created this thread so you could share them

**Comment by abao1999** [comment]
Thanks for making this thread! In addition to the CMS and ATLAS open data portals, if you're specifically interested in jets, I know of Javier Duarte at UCSD who does this: https://jduarte.physics.ucsd.edu/resources.html

^ That link has a bunch of his resources. Some of them are outdated, since no one cares about GANs anymore. But generally, pretty solid roadmap of what to do to get started. One of his main focuses is jet tagging and geometric learning (like Graph Neural Networks) for CMS/ATLAS. For instance, you might care about identifying jets while taking account the irregular detector geometry that will be (or maybe it already has been) introduced in the endcaps for high-luminosity LHC once it restarts (has it restarted yet?).

On that thread, here is FNAL LPC HATS tutorial on differentiating W bosons from the QCD background, but in CMS (If I remember correctly, ppl care about identifying the hadronic case, the semi-leptonic case, and the fully leptonic case, and the semi-leptonic one is the toughest... And also different decays of Higgs in to WW, bbbar, etc, and the di-Higgs resonance signatures)

https://github.com/FNALLPC/machine-learning-hats

These too, but maybe the're not too helpful
https://github.com/farakiko/wuhan_DL_labs/blob/master/top-tagging/TopTagging_1.ipynb
https://github.com/farakiko/wuhan_DL_labs/blob/master/top-tagging/TopTagging_2.ipynb
A data analysis challenge from CMS data school 2020: https://github.com/riga/cmsdas2020_ml/blob/master/exercise.ipynb

Lastly, I knew a guy in undergrad who wrote this paper on jet identification using these things called interaction networks (I don't think they're used much anymore, but it's from the same ppl who are big in graph/geometric learning): https://arxiv.org/pdf/1908.05318.pdf
That paper, "JEDI-NET" is all about identifying jets, and you may want to check out the datasets they use.

**Comment by abao1999** [comment]
JEDI-NET code: https://github.com/jmduarte/JEDInet-code

Looks like 2023 LPC HATS also had two workshop things on Jets https://lpc.fnal.gov/programs/schools-workshops/hats.shtml

Unfortunately, I could not see the details because I no longer have an account

**Comment by williamgilpin** [comment]
Hi, would you guys mind posting (maybe using a box link) a sample dataset so I can get an idea what it looks like? The Jupyter notebooks anthony linked look like image data, but from my understanding your MC simulations store data in a tree file.

I was looking through this PhD thesis
https://cds.cern.ch/record/2701341/files/TS2019_026.pdf

And came across this passage below---is this the same kind of data, where it can be thought of as a very sparse matrix, but it's stored as a tree to save space, and because there is some sort of connection among features that can be exploited for compression

<img width="856" alt="Screen Shot 2023-10-14 at 2 48 41 PM" src="https://github.com/williamgilpin/cphy/assets/8154246/947345b1-b505-4219-812c-c168acad3b22">

**Comment by abao1999** [comment]
@laurenlarson and all, the datasets on this page may be of interest to y'all: https://mlphysics.ics.uci.edu/

**Comment by williamgilpin** [comment]
https://lightgbm.readthedocs.io/en/stable/

https://xgboost.readthedocs.io/en/stable/

**Comment by williamgilpin** [comment]
Feature importance 
https://lightgbm.readthedocs.io/en/stable/pythonapi/lightgbm.LGBMRanker.html

**Comment by williamgilpin** [comment]
Standardize input features for any gradient-based model like a DNN

**Reply by williamgilpin** [reply]
Learn standarization on train only
